### PR TITLE
Bump min sdk version and cancel notification for disconnected calls

### PR DESCRIPTION
**Pull request by mhuynh5757**
## Submission Checklist

 - [x] Updated the `CHANGELOG.md` to reflect any **feature**, **bug fixes**, or **known issues** made in the source code
 - [x] Tested code changes and observed expected behavior in the example app
 - [x] Performed a visual inspection of the `Files changed` tab prior to submitting the pull request for review to ensure proper usage of the style guide

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

## Description

This PR bumps the `minSdkVersion` to `23` to match latest RN requirements. This PR also implements a fix for cancelling active call notifications when a call is disconnected.

## Breakdown

- Bump `minSdkVersion` to `23`.
- Cancel call notification for disconnected calls.

## Validation

- Manual testing with example app.
```
device running android api 34
outgoing call
  accepted
    [x] disconnected by remote
    [x] disconnected by local js
    [x] disconnected by local native
  [x] rejected by remote
  cancelled by local
    [x] js
    [x] native
incoming call
  closed
    [x] tap on notification body
    accepted by local native
      [x] disconnected by local js
      [x] disconnected by local native
    [x] rejected by local native
    [x] cancelled by remote
  background
    [x] tap on notification body
    accepted by local native
      [x] disconnected by local js
      [x] disconnected by local native
    [x] rejected by local native
    [x] cancelled by remote
  foreground
    accepted by local
      native
        [x] disconnected by local native
        [x] disconnected by local js
        [x] disconnected by remote
      js
        [x] disconnected by local native
        [x] disconnected by local js
        [x] disconnected by remote
    rejected by local
      [x] native
      [x] js
    [x] cancelled by remote
```

## Additional Notes

N/A

### CODE REVIEW

**Review by afalls-twilio — CHANGES_REQUESTED** [review]
This is not a good fix.. it does not bring the service back to the background

File: android/src/main/java/com/twiliovoicereactnative/VoiceService.java [hunk 1]
```diff
@@ -358,11 +358,14 @@ private void createOrReplaceForegroundNotification(final int notificationId,
...
   private void removeNotification(final int notificationId) {
+    logger.debug("removeNotification");
     NotificationManager mNotificationManager =
       (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
     mNotificationManager.cancel(notificationId);
   }
-  private void removeForegroundNotification() {
+  private void removeForegroundNotification(final int notificationId) {
+    logger.debug("removeForegroundNotification");
+    removeNotification(notificationId);
```
Comment (afalls-twilio) [inline android/src/main/java/com/twiliovoicereactnative/VoiceService.java:368]: This shouldn't be needed

Comment (mhuynh5757) [reply android/src/main/java/com/twiliovoicereactnative/VoiceService.java]: Gotcha, thanks for the assist.

**Review by afalls-twilio — APPROVED** [review]
LGTM

**Pull request merged**